edx_solutions_api_integration/mobile_api/views.py

```diff
@@ -0,0 +1,48 @@
+"""
+Views for mobile APIs
+"""
+
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+from rest_framework.exceptions import PermissionDenied
+
+from edx_solutions_api_integration.courseware_access import get_course_key, course_exists
+from edx_solutions_api_integration.models import APIUser as User
+from edx_solutions_api_integration.permissions import MobileListAPIView
+from edx_solutions_api_integration.users.views import (
+    UsersSocialMetrics,
+    UsersOrganizationsList,
+)
+from openedx.core.lib.api.permissions import IsStaffOrOwner
+from student.models import CourseEnrollment
+
+
+class MobileUsersOrganizationsList(MobileListAPIView, UsersOrganizationsList):
+    """
+    View to return list of organizations a user belongs to.
+    """
+    def __init__(self):
+        self.permission_classes += (IsStaffOrOwner, )
+
+
+class MobileUsersDiscussionMetrics(MobileListAPIView, UsersSocialMetrics):
+    """
+    View to return user discussion metrics and engagement score.
+    """
+    def get(self, request):
+        course_id = self.request.query_params.get('course_id', None)
+        username = self.request.query_params.get('username', None)
+        course_key = get_course_key(course_id)
+
+        if not course_exists(request, self.request.user, course_id):
+            raise Http404
+
+        user = self.request.user
+        if not user.is_staff and not CourseEnrollment.is_enrolled(user, course_key):
```

Comment:
@mtyaka alot of this logic can be eliminated using this [Permission class](https://github.com/edx-solutions/api-integration/blob/master/edx_solutions_api_integration/permissions.py#L90)